ooni/tests/mocks.py

```diff
@@ -242,3 +242,20 @@ def isSupported(self):
 
     def isReachable(self):
         return defer.succeed(True)
+
+
+class MockDnsResponsePayload(object):
+    def dottedQuad(self):
+        return '127.0.0.1'
+
+
+class MockDnsResponse(object):
+    payload = MockDnsResponsePayload()
+
+
+@defer.inlineCallbacks
+def mock_dns_lookup_address():
```

Comment:
Had to update the test code a bit but fortunately made it more simple. As the original function return a generator and test expect that returning a defer.succeed didn't work so i've just made it a plain python generator that works fine.